laws/src/main/scala/cats/laws/discipline/arbitrary.scala

```diff
@@ -415,6 +415,12 @@ object arbitrary extends ArbitraryInstances0 with ScalaVersionSpecific.Arbitrary
 
   implicit val catsLawsArbitraryForMiniInt: Arbitrary[MiniInt] =
     Arbitrary(Gen.oneOf(MiniInt.allValues))
+
+  implicit def catsLawsArbitraryForHashSet[A](implicit A: Arbitrary[A], hash: Hash[A]): Arbitrary[HashSet[A]] =
+    Arbitrary(getArbitrary[List[A]].map(HashSet.fromSeq(_)(hash)))
```

Comment:
I would rather see a different way to build a hashset. My opinion is basically this should have all the fundamental operations:

1. empty
2. fromSeq
3. `recurse + a`
4. `recurse - a`
5. `recurse0 | recurse1`
6. `recurse0 & recurse1`
7. `recurse0 -- recurse1` 

something like this (but you will need to use `Gen.frequency` to make sure the probability we branch into two is low enough that this doesn't go on forever.

The idea here is to simulate all the paths of creating a HashSet to make sure there aren't some buggy paths.